p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/transport/SharedHttpCacheStorage.java

```diff
@@ -89,7 +89,8 @@ protected boolean removeEldestEntry(final Map.Entry<File, CacheLine> eldest) {
      *             if the URI is know to be not found
      */
 	@Override
-	public CacheEntry getCacheEntry(URI uri, Logger logger) throws FileNotFoundException {
+	public CacheEntry getCacheEntry(URI uriParam, Logger logger) throws FileNotFoundException {
+		URI uri = uriParam.normalize();
```

Comment:
```suggestion
		URI normalized = uriParam.normalize();
```

Reply:
Do you want that rename? because then we need a few more changes (everywhere that "uri" is used)

Reply:
i pushed a rename

Reply:
Yes even if that requires more changes it should make the code more clear what URI is actually used here!